### PR TITLE
Separate out matchers

### PR DESCRIPTION
**Pull request by vagishagupta23**
Optimization Separate out matchers

Currently, all the code for matchers happens to be in main.py
which makes it quite cluttered. A separate `matchers.py` module
is created which contains all the code for matching rules and
integrated with main.py. Also the modules `test_main.py` and
`test_matchers.py` are modified accordingly.

Make sure these boxes are signed before submitting your Pull Request
-- thank you.

- [x] I have read the contributing guide lines at
  https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation
  contribution agreement at
  https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the
  changes made (if applicable)

Link
to
[redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:https://redmine.openinfosecfoundation.org/issues/2873

Describe changes:
-
-
-

### CODE REVIEW

File: suricata/update/matchers.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+    def match(self, rule):
+        if rule["noalert"]:
+            return False
+        return self.matcher.match(rule)
+
+    def filter(self, rule):
+        drop_rule = suricata.update.rule.parse(re.sub("^\w+", "drop", rule.raw))
+        drop_rule.enabled = rule.enabled
+        return drop_rule
```
Comment (shivan1b) [inline suricata/update/matchers.py]: A lot of trailing whitespace here.

File: suricata/update/matchers.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# version 2 along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+# This module contains functions for matching rules for disabling,
```
Comment (shivan1b) [inline suricata/update/matchers.py]: Trailing whitespace.

File: suricata/update/matchers.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+import re
+import os.path
+import logging
+import shlex
+import fnmatch
+import suricata.update.rule
+
+logger = logging.getLogger()
+
```
Comment (shivan1b) [inline suricata/update/matchers.py]: Leave two lines after import.

**Comment by vagishagupta23** [comment]
> @vagishagupta23 seems to be working OK but since we are creating a separate file, it would be good to fix PEP8 issues in it.

Thanks for reviewing, will take care and address these changes in the next PR.

**Comment by vagishagupta23** [comment]
Replaced by #159

**Pull request closed without merging**